circle-mlir/circle-mlir/tools/onnx2circle/src/driverDebug.cpp

```diff
@@ -72,8 +79,63 @@ static llvm::cl::opt<bool> OptCheckDynShapeInf("check_dynshapeinf",
 
 } // namespace opts
 
+class SingleRun
+{
+public:
+  static void Ensure(void)
+  {
+    SingleRun::_lock_fd = -1;
+    int rc = -1;
+    int retry = 100; // retry for 10 seconds
+
+    do
+    {
+      rc = -1;
+      SingleRun::_lock_fd = open(_lock_file, O_CREAT | O_RDWR, 0660);
+      if (_lock_fd > 0)
```

Comment:
Actually, a valid file descriptor can be 0. The standard check is to see if the descriptor is not `-1` (i.e. `_lock_fd >= 0`), so by checking `_lock_fd >` 0 the code might inadvertently ignore the case where open returns 0, which is valid.

Reply:
As I know, FD for stdin, stdout, stderr takes 0, 1 and 2.
This made me use `>` instead of `>=`.
I'll update to use `>=`